app/internal/weekly_tasks.py

```diff
@@ -0,0 +1,179 @@
+from datetime import date, datetime, time
+from typing import Iterator
+
+from app.database.models import User, Task, WeeklyTask
+from sqlalchemy.orm.session import Session
+
+
+def check_inputs(days: str, the_time: time, title: str) -> bool:
+    """Checks inputs, used by the weekly_task_from_input function"""
+    return days and the_time and title
+
+
+def weekly_task_from_input(
+    user: User,
+    title: str, days: str,
+    content: str, the_time: time,
+    is_important: bool,
+    weekly_task_id: int = 0
+) -> WeeklyTask:
+    """This function is being used to make a Weekly Task model
+        from the inputs.
+
+    Args:
+        user (User): The user who wants to make or edit a Weekly Task.
+        title (str): Title of the Weekly Task.
+        days (str): Return days of the Weekly Task.
+        content (str): Content of the Weekly Task.
+        the_time (time): Return time of the Weekly Task.
+        is_important (bool): If the task is important.
+        weekly_task_id (int): The id of the weekly task, zero if not mentioned.
+
+    Returns:
+        WeeklyTask: the model WeeklyTask which the function managed to make.
+    """
+    weekly_task = WeeklyTask(
+        title=title,
+        content=content,
+        is_important=is_important,
+        owner_id=user.id
+    )
+
+    if weekly_task_id != 0:
+        weekly_task.id = weekly_task_id
+
+    inputs_ok = check_inputs(days, the_time, title)
+    if not inputs_ok:
+        return weekly_task
+    weekly_task.days = days
+    weekly_task.the_time = the_time.strftime("%H:%M")
+    return weekly_task
+
+
+def create_weekly_task(
+    user: User, weekly_task: WeeklyTask,
+    session: Session
+) -> bool:
+    """This function is being used to add a Weekly Task to the user.
+
+        Args:
+            user (User): The user who wants to add the Weekly Task.
+            session (Session): The session to redirect to the database.
+            weekly_task (WeeklyTask): The Weekly Task that the user will add.
+
+        Returns:
+            bool: Shows if the weekly_task has been added to the db.
+        """
+    if weekly_task.days is None or weekly_task.the_time is None:
+        return False
+    user_titles = (
+        user_weekly_task.title
+        for user_weekly_task in user.weekly_tasks
+    )
+    if weekly_task.title in user_titles:
+        return False
```

Comment:
I think it's better to leave it that way. The goal is for the weekly tasks to create tasks each week, meaning they can have a maximum of 7 tasks created per week.
So it's better to have a different title for each one. So that the user can identify between them.

So the user can call his meals: lunch and dinner. or eat 1, eat 2

Reply:
I'm not sure I see the benefits. If I want to have 2 tasks in the same day? Why shouldn't I be able to do it?